06-building-a-program-start-to-finish/writing-tests/output/inverted-index.spec.js

```diff
@@ -0,0 +1,98 @@
+const InvertedIndex = require('../js/inverted-index');
+const model = require('./testmodels.js');
+
+describe('Test suite for Inverted Index Class', () => {
+  
+  const InvertedIndexTest = new InvertedIndex();
+  
+  describe('Should test for the instance of InvertedIndexTest', () => {
+    it('Should return true if InvertedIndexTest is an instance of Object', () => {
+      expect(InvertedIndexTest instanceof Object).toBe(true);
+    });
+    
+    it('Should return false if InvertedIndexTest is not an instance of Array', () => {
+      expect(InvertedIndexTest instanceof Array).toBe(false);
+    });
+    
+    it('Should return true if InvertedIndexTest is an instance of InvertedIndex class', () => {
+      expect(InvertedIndexTest instanceof InvertedIndex).toBe(true);
+    });
+    
+    it('Should return false if InvertedIndexTest is not an instance of FakeInvertedIndex class', () => {
+      expect(InvertedIndexTest instanceof model.FakeInvertedIndex).toBe(false);
+    });
+  });
+
+  
+  describe('Should test for validity of data', () => {
+    it('should return true on valid json data', () => {
+      expect(InvertedIndexTest.isValid(model.data)).toBe(true);
+    });
+    
+    it('should return false on invalid json data', () => {
+      expect(InvertedIndexTest.isValid(model.fake_data)).toBe(false);
```

Comment:
I like that you have a model where you store data for tests, I'd like to see you use better names for these objects.
For example, the spec on line 32 is testing for `invalid json data` but the model being uses is named `fake_data`. It'll help to have descriptive names.